### DISCUSSION TITLE
How to add custom overrides programmatically?

### DISCUSSION THREAD

**Discussion by jlkiri** [BurntSushi/ripgrep#1662] [General] [answered]
It seems that `overrides` method on `WalkBuilder` should be used but I can't define an `Override` tuple because its field (`Gitignore`) is not public.

**Comment by BurntSushi** [comment, marked as answer]
You need to use an [`OverrideBuilder`](https://docs.rs/ignore/0.4.16/ignore/overrides/struct.OverrideBuilder.html).

**Reply by jlkiri** [reply]
Oh, thank you. For some reason I didn't notice it in the docs.

**Reply by BurntSushi** [reply]
Yeah, the docs for `ignore` are quite poor unfortunately. The entire crate isn't great. Its internals are complicated. Its API is complicated. Docs are sparse. There are precious few examples. It needs a lot of love and I just haven't had the time to give it.
